### PR TITLE
(PUP-7326) Add ADSI / Principal localizations

### PR DESCRIPTION
**Pull request by Iristyle**
- Add _() calls for localizable strings, so that it will be easier
   after the next merge (which will have a few easily resolved
   merge conflicts) to perform automated merge-ups

### CODE REVIEW

**Comment by Iristyle** [comment]
This will still cause a few minor merge conflicts that must be resolved (no wary around that), but on merge-up, we can now apply the strategy to take the 4.10.x branch changes.

File: lib/puppet/util/windows/principal.rb [hunk 1]
```diff
@@ -116,14 +117,14 @@ def self.lookup_account_sid(system_name = nil, sid_bytes)
                 last_error = FFI.errno
 
                 if (success == FFI::WIN32_FALSE && last_error != ERROR_INSUFFICIENT_BUFFER)
-                  raise Puppet::Util::Windows::Error.new("Failed to call LookupAccountSidW with bytes: #{sid_bytes}", last_error)
+                  raise Puppet::Util::Windows::Error.new(_('Failed to call LookupAccountSidW with bytes: %{sid_bytes}') % { sid_bytes: sid_bytes}, last_error)
```
Comment (glennsarti) [inline lib/puppet/util/windows/principal.rb]: Needs translator note here for literal `LookupAccountSidW`

File: lib/puppet/util/windows/principal.rb [hunk 2]
```diff
@@ -116,14 +117,14 @@ def self.lookup_account_sid(system_name = nil, sid_bytes)
...
                 end
 
                 FFI::MemoryPointer.new(:lpwstr, name_length_ptr.read_dword) do |name_ptr|
                   FFI::MemoryPointer.new(:lpwstr, domain_length_ptr.read_dword) do |domain_ptr|
                     if LookupAccountSidW(system_name_ptr, sid_ptr, name_ptr, name_length_ptr,
                         domain_ptr, domain_length_ptr, name_use_enum_ptr) == FFI::WIN32_FALSE
-                     raise Puppet::Util::Windows::Error.new("Failed to call LookupAccountSidW with bytes: #{sid_bytes}")
+                     raise Puppet::Util::Windows::Error.new(_('Failed to call LookupAccountSidW with bytes: %{sid_bytes}') % { sid_bytes: sid_bytes} )
```
Comment (glennsarti) [inline lib/puppet/util/windows/principal.rb]: Needs translator note here for literal `LookupAccountSidW`

**Comment by Iristyle** [comment]
Per @LarissaLane translator notes are not necessary here, so this should be ready to merge if there are no other issues.

**Pull request merged**